### PR TITLE
Add limit to comprehensions

### PR DESCRIPTION
**Pull request by AlexJuca**
This PR implements :limit option for comprehensions to restrict the amount of results returned.

### CODE REVIEW

File: lib/elixir/src/elixir_expand.erl [hunk 1]
```diff
@@ -702,17 +702,21 @@ generated_case_clauses([{do, Clauses}]) ->
...
 
-validate_for_options([{into, _} = Pair | Opts], _Into, Uniq, Reduce) ->
-  validate_for_options(Opts, Pair, Uniq, Reduce);
-validate_for_options([{uniq, Boolean} = Pair | Opts], Into, _Uniq, Reduce) when is_boolean(Boolean) ->
-  validate_for_options(Opts, Into, Pair, Reduce);
-validate_for_options([{uniq, Value} | _], _, _, _) ->
+validate_for_options([{into, _} = Pair | Opts], _Into, Uniq, Reduce, Limit) ->
+  validate_for_options(Opts, Pair, Uniq, Reduce, Limit);
+validate_for_options([{limit, Integer} = Pair | Opts], Into, _Uniq, Reduce, Limit) when is_integer(Integer) ->
+validate_for_options(Opts, Into, Pair, Reduce, Limit);
```
Comment (josevalim) [inline lib/elixir/src/elixir_expand.erl:708]:
```suggestion
  validate_for_options(Opts, Into, Pair, Reduce, Limit);
```

Comment (AlexJuca) [reply lib/elixir/src/elixir_expand.erl]: Looks good.

**Comment by benwilson512** [comment]
At the risk of bike shedding, would `take` as the option name be better since it is serving the same function as the already existing `Stream.take/2` and `Enum.take/2`? If there was a mailing list thread that went through this I apologize for missing it!=

**Comment by josevalim** [comment]
The reason why I avoided `take` is because you have an impression that the take happens after the computation (although that's not generally true for Stream.take). `limit` is clearer that no further computations or items will be taken once you reach the limit. But maybe we should indeed stick with take.

**Comment by benwilson512** [comment]
> that the take happens after the computation
 
I hear that concern. The thing is, we already have to educate folks with `Stream.take` and `Enum.take` that this does not happen. If we keep the `for` option as `take` we get to go "take halts computation when it has enough" as a generic rule, vs "for `:limit` is the same as `Stream/Enum.take`, they stop computation when it has enough".

**Comment by josevalim** [comment]
Ack, I will update the original issue, thank you.

Also closing this as it has gone stale with this change.

**Pull request closed without merging**